### PR TITLE
Add Bun.sh with good defaults.

### PR DESCRIPTION
**Pull request by jtwebman**
This adds Bun.sh 1.0.0 to the mix for TypeScript and JavaScript

### CODE REVIEW

File: frameworks/TypeScript/bun/benchmark_config.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        "database": "Postgres",
+        "framework": "bun",
+        "language": "Typescript",
+        "flavor": "None",
+        "orm": "Micro",
+        "platform": "Bun",
+        "webserver": "Elysia",
+        "os": "Linux",
+        "database_os": "Linux",
+        "display_name": "Bun Elysia Posgres",
```
Comment (muuvmuuv) [inline frameworks/TypeScript/bun/benchmark_config.json:24]: Postgres* :)

File: frameworks/TypeScript/bun/bun.dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
+FROM debian:bookworm as build
```
Comment (muuvmuuv) [inline frameworks/TypeScript/bun/bun.dockerfile:1]: There is an official docker image which might fit better? https://hub.docker.com/r/oven/bun

Comment (jtwebman) [reply frameworks/TypeScript/bun/bun.dockerfile]: Switch

**Comment by marcbarbosa** [comment]
@jtwebman Since you are using a specific framework (Elysia), I believe we should move your implementation to `frameworks/TypeScript/Elysia/` directory.

**Comment by nbrady-techempower** [comment]
Yes, please move this to `frameworks/TypeScript/elysia` and update the `package.json` to reflect. `frameworks/TypeScript/bun` would be reserved for a "vanilla" bun implementation without using another web framework.

**Comment by nbrady-techempower** [comment]
closing in favor of #8480

**Pull request closed without merging**